packages/loader/container-loader/src/container.ts

```diff
@@ -162,6 +162,80 @@ export enum ConnectionState {
     Connected,
 }
 
+enum LifecycleState {
+    /**
+     * The container has been created
+     */
+    Created = "created",
+
+    /**
+     * The container is loading
+     */
+    Loading = "loading",
+
+    /**
+     * The container has been loaded
+     */
+    Loaded = "loaded",
+
+    /**
+     * The container has been marked for closure
+     */
+    Closing = "closing",
+
+    /**
+     * The container is in closed state
+     */
+    Closed = "closed",
+}
+
+class LifecycleStateHandler {
+    constructor(state: LifecycleState) {
```

Comment:
I think let's not take the state and always initialize to `Created`

Reply:
Done